### PR TITLE
update gui/cp437-table to use modern widgets

### PR DESCRIPTION
**Pull request by myk002**
movable window, widgets to manage rendering, state, and text cursor

depends on DFHack/dfhack#2551

### CODE REVIEW

File: gui/cp437-table.lua [hunk 1]
```diff
@@ -1,118 +1,95 @@
...
-                if ch then
-                    self:insert(ch)
-                end
-            end
+function CPDialog:onInput(keys)
+    local x, y = self.subviews.board:getMousePos()
+    if keys._MOUSE_L_DOWN and x then
+        local ch = x + (32 * y)
+        if ch ~= 0 and dfhack.screen.charToKey(ch) then
+            self.subviews.edit:insert(string.char(ch))
```
Comment (lethosor) [inline gui/cp437-table.lua:80]: Where's this method defined? EditField doesn't have an insert() method on develop

Comment (myk002) [reply gui/cp437-table.lua:80]: oops, I may not have made a PR for that yet. One second.

Comment (myk002) [reply gui/cp437-table.lua:80]: DFHack/dfhack#2551

**Review by lethosor — APPROVED** [review]
Seems to look and behave the same way as before

**Pull request merged**